tests/conftest.py

```diff
@@ -22,6 +23,47 @@
 FM_NQUBITS = 2
 
 
+@fixture
+def noisy_pulser_sim() -> Tensor:
```

Comment:
The test cases are not uniform, we have density matrix comparison for one and the expectation value for the other. Why is it done this way?  
Also are we just storing the output from actual simulation and comparing with it again ? The test case was  just to check if the workflow goes through smoothly. Is the assertion check in the end important ?

Reply:
Addressed in #427